### PR TITLE
Added date formatting according to chosen language in the settings

### PR DESCRIPTION
**Pull request by artsiom-voitas**
This pull requests addresses issue [#2406](https://github.com/spacedriveapp/spacedrive/issues/2406)
- added an option in settings to choose preferred way for displaying date
- added i18n entries and translations for format date settings
- added English (UK) language to distinguish Date Format on English between US variant and European

### CODE REVIEW

**Comment by artsiom-voitas** [comment]
> Code looks good, will test it soon

Please check if the language of the displayed date format always changes when the language is changed. I have a situation where with 3-4 language switches in a row, the date displayed in the settings is not always updated to the language of the date format in the selector (I think I've made a mistake somewhere, and it doesn't always trigger a component reload, but I don't see where). It would be great to fix this issue before adding the function to the main branch. Thank you for your help

**Comment by utkubakir** [comment]
> Please check if the language of the displayed date format always changes when the language is changed. I have a situation where with 3-4 language switches in a row, the date displayed in the settings is not always updated to the language of the date format in the selector (I think I've made a mistake somewhere, and it doesn't always trigger a component reload, but I don't see where). It would be great to fix this issue before adding the feature to the main branch. Thank you for your help

Looks like a race condition to me, setting the language format after i18n seems to be more consistent

**Review by utkubakir — APPROVED** [review]

File: interface/app/$libraryId/Explorer/util.ts [hunk 1]
```diff
@@ -33,3 +34,58 @@ export function getItemId(index: number, items: ExplorerItem[]) {
...
+	it: () => import('dayjs/locale/it.js'),
+	ja: () => import('dayjs/locale/ja.js')
+};
+
+export function loadDayjsLocale(language: string) {
+	dayjsLocales[language]().then(() => {
+		language = language.replace('_', '-');
+		dayjs.locale(language);
+	});
+}
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/util.ts:59]: can this be optimized? bundling all locales doesn't sound smart

**Review by utkubakir — APPROVED** [review]

**Comment by utkubakir** [comment]
Thank you, lgtm @artsiom-voitas

**Pull request merged**